getting started/crowdsale.sol

```diff
@@ -1,73 +1,61 @@
-contract token { 
-    mapping (address => uint) public balance; 
-    function token() {}  
-    function sendToken(address receiver, uint amount) returns(bool sufficient) {  } 
-}
 
-contract CrowdSale {
+    contract token { mapping (address => uint) public coinBalanceOf; function token() {}  function sendCoin(address receiver, uint amount) returns(bool sufficient) {  } }
     
-    address public admin;
-    address public beneficiary;
-    uint public fundingGoal;
-    uint public amountRaised;
-    uint public deadline;
-    uint public price;
-    token public tokenReward;   
-    Funder[] public funders;
-    
-    /* data structure to hold information about campaign contributors */
-    struct Funder {
-        address addr;
-        uint amount;
-    }
-    
-    /*  at initialization, setup the owner */
-    function CrowdSale() {
-        admin = msg.sender;
-    }   
-    
-    function setup(address _beneficiary, uint _fundingGoal, uint _duration, uint _price, address _reward) returns (bytes32 response){
-        if (msg.sender == admin && !(beneficiary > 0 && fundingGoal > 0 && deadline > 0)) {
+    contract CrowdSale {
+        
+        address public beneficiary;
+        uint public fundingGoal; uint public amountRaised; uint public deadline; uint public price;
+        token public tokenReward;   
+        Funder[] public funders;
+        event FundTransfer(address backer, uint amount, bool isContribution);
+        
+        /* data structure to hold information about campaign contributors */
+        struct Funder {
+            address addr;
+            uint amount;
+        }
+        
+        /*  at initialization, setup the owner */
+        function CrowdSale(address _beneficiary, uint _fundingGoal, uint _duration, uint _price, address _reward) {
```

Comment:
You can directly use `token _reward`, it will be converted to `address` for the ABI.